tests/init_2.f90

```diff
@@ -0,0 +1,18 @@
+program main
+    use mpi, only: MPI_Init_thread, MPI_Finalize, MPI_THREAD_FUNNELED
+    implicit none
+    
+    integer :: provided, ierr
+    
+    ! Initialize MPI with thread support
+    call MPI_Init_thread(MPI_THREAD_FUNNELED, provided, ierr)
+    
+    if (ierr /= 0) then
+        print *, "Error initializing MPI with threads"
+        stop
```

Comment:
I think we should've `error stop` here, because if we just have `stop` here, then even when `ierr /= 0` then `stop` is executed, and the program will still exit gracefully (i.e. with zero exit status), and hence the test will not fail.

Reply:
I've resolved this conversation in the newly pushed commit.